frontend/src/pages/dashboard/components/tables/trucks-table/trucks-table.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button, Modal, Table } from '~/libs/components/components.js';
+import { DataStatus, IconName } from '~/libs/enums/enums.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useAppTable,
+  useCallback,
+  useQueryParameters,
+  useState,
+} from '~/libs/hooks/hooks.js';
+import { type PaginationParameters } from '~/libs/types/types.js';
+import {
+  type TruckAddRequestDto,
+  type TruckGetAllResponseDto,
+} from '~/packages/trucks/libs/types/types.js';
+import { findAllTrucksForBusiness } from '~/slices/trucks/actions.js';
+import { actions as truckActions } from '~/slices/trucks/trucks.js';
+
+import { AddTruckForm } from '../../form/form.js';
+import { columns } from './columns/columns.js';
+import styles from './styles.module.scss';
+
+const TrucksTable: React.FC = () => {
+  const { trucks, total, dataStatus } = useAppSelector(({ trucks }) => ({
+    trucks: trucks.trucks,
+    total: trucks.total,
+    dataStatus: trucks.dataStatus,
+  }));
+  const { getQueryParameters } = useQueryParameters();
+
+  const initialSize = getQueryParameters('size') as string | null;
+  const initialPage = getQueryParameters('page') as string | null;
+
+  const dispatch = useAppDispatch();
+
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+
+  const { pageSize, pageIndex, changePageSize, changePageIndex } = useAppTable<
+    TruckGetAllResponseDto,
+    PaginationParameters
+  >({
+    tableFetchCall: findAllTrucksForBusiness,
+    initialPageIndex: initialPage ? +initialPage : undefined,
+    initialPageSize: initialSize ? +initialSize : undefined,
+  });
+
+  const handleAddTruckModalVisibility = useCallback(() => {
+    setIsModalOpen(!isModalOpen);
```

Comment:
```suggestion
    setIsModalOpen(prev => !prev);
```
and can we define useModal() hook? It can incapsulate state and toggle callback
```
 const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
const handleAddTruckModalVisibility
```

Reply:
We have this useModal hook, but it is used only in the Modal component. However, I can write something like useToogle for this purpose. Because we have a lot of such reusable code in the project. Should I write a hook?

Reply:
Yes, please 👍